src/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -102,4 +106,32 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
         }
     }
 
+    /***
+    * @description Creates/modifies schedule record(s) after insert/update of recurring donation
+    * @param rds Trigger.new records
+    * @param oldRds Trigger.old records
+    * @param triggerAction BeforeInsert, AfterInsert, etc.
+    */
+    private void handleRDSchedules(List<SObject> rds, List<SObject> oldRds, TDTM_Runnable.Action triggerAction) {
+
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
+            buildNewSchedules(rds);
+        }
+    }
+
+    /***
+    * @description Handles schedule creation after insert of recurring donation
+    * @param rds Trigger.new records
+    */
+    private void buildNewSchedules(List<npe03__Recurring_Donation__c> rds) {
+
+        for (npe03__Recurring_Donation__c rd: rds) {
+            dmlActions.objectsToInsert.addAll(
+                    new RD2_ScheduleService()
```

Comment:
Since service methods need to be independent, we can call one method instead. Would the following work wrt future requirements and changes?
```
RD2_ScheduleService scheduleService = new RD2_ScheduleService();

for (npe03__Recurring_Donation__c rd : rds) {
    dmlActions.objectsToInsert.addAll(scheduleService.getNewSchedules(rd));  
}
```